tests/test_web.py

```diff
@@ -241,5 +232,31 @@ def test_email_body_was_sent_with_expected_body(
     assert server.sendmail.call_args_list == [
         call(ANY, ANY, ANY)
     ]
-    email_content = server.sendmail.call_args.args[2]
+    email_content = server.sendmail.call_args.args[MESSAGE_ARGUMENT]
     assert expected_body in email_content
+
+
+@pytest.mark.parametrize(
+    "environment",
+    [
+        (
+            {
+                "INVOICE_UTILS_MAIL_SUBJECT": "test subject",
+                "INVOICE_UTILS_BODY_TEMPLATE_PATH": "test_template.html",
+                "INVOICE_UTILS_SENDER_EMAIL": "test@email.com"
+            }
+        )
+    ],
+    indirect=["environment"]
+)
+def test_sendmail_was_called_with_pdf_attachment(environment, http, server, email_invoice_request_body):
```

Comment:
Does the test generate a file on disk? If so, find a way that it doesn't. Unit tests need to be able to run in isolation, on platforms that don't have disk access.
I suggest you have a look at our good friends `patch` and `MagicMock`.

Reply:
I patched the render method to save time, although in the end I chose to not create any files at this point as it's not necessary for now.